pallets/order-book/src/lib.rs

```diff
@@ -239,51 +241,55 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type OrderIdNonceStore<T: Config> = StorageValue<_, T::OrderIdNonce, ValueQuery>;
 
-	/// Map of Vec containing OrderIds of same asset in/out pairs.
+	/// Map of Vec containing OrderIds of same currency in/out pairs.
 	/// Allows looking up orders available corresponding pairs.
 	///
 	/// NOTE: The key order is (currency_in, currency_out).
 	#[pallet::storage]
-	pub type AssetPairOrders<T: Config> = StorageDoubleMap<
+	pub type CurrencyPairOrders<T: Config> = StorageDoubleMap<
 		_,
 		Twox64Concat,
-		T::AssetCurrencyId,
+		T::CurrencyId,
 		Twox64Concat,
-		T::AssetCurrencyId,
+		T::CurrencyId,
 		BoundedVec<T::OrderIdNonce, T::OrderPairVecSize>,
 		ValueQuery,
 	>;
```

Comment:
Do we still need that?

Reply:
Can't we make it an `NMap`? Would make it a lot easier and we do not have the size restriction anymore.

Reply:
> Do we still need that?

TBH, I do not know. Why did we need it before? I understand that for Apps, it's an easy way to look for existing orders to fill. But not sure if it used now.

> Can't we make it an NMap? 

Makes more sense to me. Also, I would add the ratio to it: `MMap<(CurrencyID, CurrencyID, OrderRatio), OrderId>` i.e. to look for orders that use market ratios.

But by now, if not used by Apps (@onnovisser), I would just remove this, and I'll add it again once we know exactly how we want to index it, WDYT?

Reply:
If we assume we will only have a reasonable amount of orders in paralel that will be fine. The UI will most likely need to filter based on currency, but if they can just fetch all orders and sort them accordingly that should be fine I ugess.

Reply:
the queries that the UI uses are `orderBook.orders` and `orderBook.tradingPair`. Because we also want to filter by outCurrency and can't otherwise do that, we just get all orders and filter them in the UI, and assume they will be a reasonable amount like Frederik said.

Reply:
Ok, so I would remove this by now

Reply:
In case of aggrement, here: [e579147](https://github.com/centrifuge/centrifuge-chain/pull/1674/commits/e57914785cafc142341c82d160d05e3292bb24ff), If not I revert